lib/rule-tester/rule-tester.js

```diff
@@ -1012,6 +1023,13 @@ class RuleTester {
                                     rehydratedMessage,
                                     `Hydrated message "${rehydratedMessage}" does not match "${message.message}"`
                                 );
+                            } else {
```

Comment:
I think we should check this even if there is `data` property in the test error.

It would catch cases like this:

```js
messages: {
    unexpected: "Unexpected {{nam}}." // typo name -> nam
}

context.report({
    node,
    messageId: "unexpected",
    data: { name }
});

// this test is passing, although the message is  "Unexpected {{nam}}." literally.
{
    code: "alert(foo)",
    errors: [{ messageId: "unexpected", data: { name: "alert" }]
}
```